### PR TITLE
fix(ingest): bigquery - Fixing querying non-date partition columns in profiling

### PR DESCRIPTION
**Pull request by treff7es**
Adding to the source report the number of non-eligible tables for profiling

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/profiler.py [hunk 1]
```diff
@@ -115,11 +115,17 @@ def generate_partition_profiler_query(
...
                     return None, None
 
+                partition_column_type: str = "DATE"
+                for c in table.columns:
+                    if c.is_partition_column:
+                        partition_column_type = c.data_type
+
                 if table.time_partitioning.type_ in ("DAY", "MONTH", "YEAR"):
-                    partition_where_clause = "{column_name} BETWEEN DATE('{partition_id}') AND DATE('{upper_bound_partition_id}')".format(
+                    partition_where_clause = "{column_name} BETWEEN {partition_column_type}('{partition_id}') AND {partition_column_type}('{upper_bound_partition_id}')".format(
```
Comment (szalai1) [inline metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/profiler.py:124]: suggestion: you could use f-strings here and avoid `.format`

File: metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/profiler.py [hunk 2]
```diff
@@ -216,14 +222,10 @@ def get_bigquery_profile_request(
...
-            # Profile only table level if dataset is filtered from profiling
-            # due to size limits alone
-            if self.is_dataset_eligible_for_profiling(
-                dataset_name, table.last_altered, 0, 0
-            ):
-                profile_table_level_only = True
-            else:
-                skip_profiling = True
+            profile_table_level_only = True
+            self.report.num_tables_not_eligible_profiling[dataset] = (
```
Comment (szalai1) [inline metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/profiler.py:226]: suggestion: make `num_tables_not_eligible_profiling` a Counter and no need to do this tricks.
just: `self.report.num_tables_not_eligible_profiling[dataset]++`
https://docs.python.org/3/library/collections.html

File: metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/usage.py [hunk 3]
```diff
@@ -66,7 +66,7 @@ class OperationalDataMeta:
 def bigquery_audit_metadata_query_template(
     dataset: str,
     use_date_sharded_tables: bool,
-    table_allow_filter: str = None,
+    table_allow_filter: Optional[str] = None,
```
Comment (szalai1) [inline metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/usage.py:69]: was this change necessary?

**Review by szalai1 — APPROVED** [review]

File: metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/usage.py (hunk 3 above)
Comment (treff7es) [reply metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/usage.py:69]: This field should be Optional and mypy (linter) was complaining about it.

File: metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/profiler.py (hunk 2 above)
Comment (treff7es) [reply metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/profiler.py:226]: we try to use in report structs that can't grow indefinitely and there this way I think won't work :(
